src/kernel/bitcoinkernel.h

```diff
@@ -0,0 +1,1036 @@
+} kernel_LoggingOptions;
+
+/**
+ * A collection of error codes that may be issued by the kernel library.
+ */
+typedef enum {
+    kernel_ERROR_OK = 0,
+    kernel_ERROR_TX_INDEX,
+    kernel_ERROR_TX_SIZE_MISMATCH,
+    kernel_ERROR_TX_DESERIALIZE,
+    kernel_ERROR_AMOUNT_REQUIRED,
+    kernel_ERROR_INVALID_FLAGS,
+    kernel_ERROR_INVALID_FLAGS_COMBINATION,
+    kernel_ERROR_SPENT_OUTPUTS_REQUIRED,
+    kernel_ERROR_SPENT_OUTPUTS_MISMATCH,
+    kernel_ERROR_INVALID_POINTER,
+    kernel_ERROR_LOGGING_FAILED,
+    kernel_ERROR_UNKNOWN_OPTION,
+    kernel_ERROR_INVALID_CONTEXT,
+    kernel_ERROR_INTERNAL,
+    kernel_ERROR_OUT_OF_BOUNDS,
+} kernel_ErrorCode;
+
+/**
+ * Contains an error code and a pre-defined buffer for containing a string
+ * describing a possible error.
+ */
+typedef struct {
+    kernel_ErrorCode code;
+    char message[256];
+} kernel_Error;
+
+/**
+ * Script verification flags that may be composed with each other.
+ */
+typedef enum
+{
+    kernel_SCRIPT_FLAGS_VERIFY_NONE                = 0,
+    kernel_SCRIPT_FLAGS_VERIFY_P2SH                = (1U << 0), //!< evaluate P2SH (BIP16) subscripts
+    kernel_SCRIPT_FLAGS_VERIFY_DERSIG              = (1U << 2), //!< enforce strict DER (BIP66) compliance
+    kernel_SCRIPT_FLAGS_VERIFY_NULLDUMMY           = (1U << 4), //!< enforce NULLDUMMY (BIP147)
+    kernel_SCRIPT_FLAGS_VERIFY_CHECKLOCKTIMEVERIFY = (1U << 9), //!< enable CHECKLOCKTIMEVERIFY (BIP65)
+    kernel_SCRIPT_FLAGS_VERIFY_CHECKSEQUENCEVERIFY = (1U << 10), //!< enable CHECKSEQUENCEVERIFY (BIP112)
+    kernel_SCRIPT_FLAGS_VERIFY_WITNESS             = (1U << 11), //!< enable WITNESS (BIP141)
+
+    kernel_SCRIPT_FLAGS_VERIFY_TAPROOT             = (1U << 17), //!< enable TAPROOT (BIPs 341 & 342)
+    kernel_SCRIPT_FLAGS_VERIFY_ALL                 = kernel_SCRIPT_FLAGS_VERIFY_P2SH |
+                                                     kernel_SCRIPT_FLAGS_VERIFY_DERSIG |
+                                                     kernel_SCRIPT_FLAGS_VERIFY_NULLDUMMY |
+                                                     kernel_SCRIPT_FLAGS_VERIFY_CHECKLOCKTIMEVERIFY |
+                                                     kernel_SCRIPT_FLAGS_VERIFY_CHECKSEQUENCEVERIFY |
+                                                     kernel_SCRIPT_FLAGS_VERIFY_WITNESS |
+                                                     kernel_SCRIPT_FLAGS_VERIFY_TAPROOT
+} kernel_ScriptFlags;
+
+/**
+ * A helper struct for a single transaction output.
+ */
+typedef struct {
+    int64_t value;
+    const unsigned char* script_pubkey;
+    size_t script_pubkey_len;
+} kernel_TransactionOutput;
+
+/**
+ * Chain type used for creating chain params.
+ */
+typedef enum {
+    kernel_CHAIN_TYPE_MAINNET = 0,
+    kernel_CHAIN_TYPE_TESTNET,
+    kernel_CHAIN_TYPE_SIGNET,
+    kernel_CHAIN_TYPE_REGTEST,
+} kernel_ChainType;
+
+typedef struct {
+    unsigned char hash[32];
+} kernel_BlockHash;
```

Comment:
I think I'd prefer not wrapping this:
```suggestion
typedef uint8_t kernel_BlockHash[32];
```

Reply:
The two have different type safety guarantees. You can pass other uint8_t arrays to a function taking an array typedef, but you get a type error if you try something similar with a different struct. I think this is worth the type safety here. In the code we just use the non-type-safe `uint256` for the block hash, but I think it would be preferable to use a struct, since we'll probably do the same for the strongly-typed transaction identifiers.